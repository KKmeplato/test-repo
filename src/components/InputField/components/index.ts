export * from "./StepperControl";
export * from "./StepperIcon";
