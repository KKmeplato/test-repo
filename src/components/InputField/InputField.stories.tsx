import { action } from "@storybook/addon-actions";
import { Meta, StoryObj } from "@storybook/react";
import { hideStorybookControls } from "development_support/utils";
import { useState } from "react";

import { InputField, InputFieldProps, InputProps } from "components/InputField";
import { Icon } from "components/icons";

const meta: Meta<typeof InputField> = {
  component: InputField,
  decorators: [
    (Story) => (
      <div style={{ width: "400px" }}>
        <Story />
      </div>
    )
  ]
};

export default meta;
type Story = StoryObj<InputProps>;
type NumberInputStory = StoryObj<Extract<InputFieldProps, { type: "number" }>>;

function TextInputWithArgs(args: InputProps) {
  const [value, setValue] = useState(args.value);

  return <InputField {...args} value={value} onChange={setValue} />;
}

export const TextInput: Story = {
  args: {
    label: "Text input field",
    hint: "Lorem ipsum dolor sit amet, consetetur",
    showHintIcon: true,
    prefix: "https://",
    suffix: "DE",
    suffixIcon: <Icon.info />,
    prefixIcon: <Icon.info />,
    showCharacterCount: true,
    disabled: false,
    preview: false,
    id: "test-input-field",
    maxLength: 10,
    type: "text",
    onChange: action("onChange")
  },
  render: TextInputWithArgs,
  argTypes: {
    ...hideStorybookControls(["type"])
  }
};

function InputFieldWithArgs(
  args: Extract<InputFieldProps, { type: "number" }>
) {
  const [value, setValue] = useState(args.value);

  return <InputField {...args} value={value} onChange={setValue} />;
}

export const NumberInput: NumberInputStory = {
  args: {
    type: "number",
    label: "Text input field",
    hint: "Lorem ipsum dolor sit amet, consetetur",
    showHintIcon: true,
    prefix: "€",
    value: 1,
    suffix: "€",
    disabled: false,
    preview: false,
    id: "test-input-field",
    min: 0,
    max: 10,
    onChange: action("onChange")
  },
  render: InputFieldWithArgs,
  argTypes: {
    type: {
      control: { type: "radio" },
      options: ["number"]
    },
    ...hideStorybookControls([
      "type",
      "prefixIcon",
      "suffixIcon",
      "showCharacterCount",
      "multiline"
    ])
  }
};
