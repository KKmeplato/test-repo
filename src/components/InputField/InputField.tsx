import React, { createElement, forwardRef, useId, useRef } from "react";

import { Icon } from "components/icons";
import { cn } from "lib/utils";

import { StepperControl } from "./components";
import type { InputFieldProps } from "./types";

export const InputField = forwardRef<HTMLInputElement, InputFieldProps>(
  (
    {
      className,
      disabled,
      error,
      hint,
      id,
      label,
      maxLength,
      multiline,
      prefix,
      prefixIcon,
      preview,
      showCharacterCount,
      showHintIcon,
      success,
      suffix,
      suffixIcon,
      value = "",
      type,
      onChange,
      ...rest
    },
    forwardedRef
  ) => {
    const generatedInputRef = useRef<HTMLInputElement>();
    const inputRef = forwardedRef || generatedInputRef;
    const isActive = !disabled && !preview;
    const hasValidationMessage = success || error;
    const generatedId = useId();
    const inputId = id || generatedId;
    const inputField = createElement(multiline ? "textarea" : "input", {
      "aria-disabled": disabled || preview,
      className: cn(
        "w-full bg-interactive-plain rounded font-normal leading-tight appearance-none text uikit-hide-number-spin-controls",
        multiline
          ? "rounded-level-2 px-3 py-2 justify-start items-center gap-1 inline-flex"
          : "focus:outline-none text-base bg-transparent",
        {
          "text-success": success,
          "text-critical": error,
          "bg-transparent text-default group-hover:bg-transparent hover:bg-transparent group-hover:disabled:bg-transparent":
            preview,
          "bg-surface-disabled text-disabled": disabled && !preview
        }
      ),
      disabled: disabled || preview,
      id,
      maxLength,
      value,
      ref: inputRef,
      onChange: (e: React.ChangeEvent<HTMLInputElement>) => {
        if (type === "number") {
          const numberValue = parseInt(e.target.value);
          return onChange?.(isNaN(numberValue) ? undefined : numberValue);
        }

        return onChange?.(e.target.value);
      },
      type,
      ...rest
    });

    const labelMarkup = label ? (
      <label
        htmlFor={inputId}
        className={cn("w-full text-sm font-medium leading-tight text", {
          "text-disabled": disabled && !preview
        })}
      >
        {label}
      </label>
    ) : null;
    const charactersCountMarkup =
      showCharacterCount && maxLength && !preview ? (
        <span
          className={cn(
            "text-right text-sm font-normal leading-tight text-subdued",
            {
              "text-disabled": disabled
            }
          )}
        >
          {`${value?.toString().length || 0}/${maxLength}`}
        </span>
      ) : null;
    const hintMarkup =
      hint && !preview && (!hasValidationMessage || disabled) ? (
        <div className="inline-flex gap-1">
          {showHintIcon && (
            <Icon.info
              className={cn("h-5 w-5 rounded-level-2 text-icon", {
                "text-icon-disabled": disabled
              })}
            />
          )}
          <p
            className={cn(
              "shrink grow basis-0 text-sm font-normal leading-5 text-subdued",
              {
                "text-disabled": disabled
              }
            )}
          >
            {hint}
          </p>
        </div>
      ) : null;
    const messageMarkup =
      hasValidationMessage && isActive ? (
        <div
          className={cn("inline-flex gap-2", {
            "text-success": success,
            "text-critical": error
          })}
        >
          <div className={"inline-flex h-5 w-5 justify-center"}>
            {success ? <Icon.success /> : <Icon.error />}
          </div>
          <div
            className={cn("text-sm font-normal", {
              "text-success": success,
              "text-critical": error
            })}
          >
            {success || error}
          </div>
        </div>
      ) : null;

    const stepperControlMarkup = type === "number" && !preview && (
      <StepperControl
        min={rest?.min as number | undefined}
        max={rest.max as number | undefined}
        value={
          (isNaN(parseInt(String(value))) ? undefined : Number(value)) as
            | number
            | undefined
        }
        disabled={disabled}
        onChange={(value?: number) => {
          if (typeof inputRef !== "function") {
            inputRef.current?.focus();
          }
          onChange?.(value);
        }}
      />
    );

    return (
      <>
        {multiline ? (
          <div
            className={cn(
              "w-full flex-col items-start justify-start gap-1",
              className
            )}
          >
            <div className="inline-flex h-5 w-full items-center justify-center gap-1 pr-1">
              {labelMarkup}
              {charactersCountMarkup}
            </div>
            {inputField}
            {hintMarkup}
            {messageMarkup}
          </div>
        ) : (
          <div
            className={cn(
              "inline-flex w-full flex-col items-start justify-start gap-1",
              className
            )}
          >
            {labelMarkup}
            <div
              className={cn(
                "group inline-flex h-9 w-full items-center justify-start gap-1 rounded-level-2 border border-solid border-default bg-interactive-plain py-2 pl-3 pr-2 focus-within:border-focus focus-within:ring-1 focus-within:ring-focus hover:border-hovered hover:bg-interactive-plain-hovered hover:focus-within:border-focus hover:focus-within:bg-interactive-plain",
                {
                  "border-disabled bg-surface-disabled text-disabled hover:border-disabled hover:bg-surface-disabled":
                    disabled && !preview,
                  "border-interactive-success focus-within:border-interactive-success focus-within:ring-1 focus-within:ring-interactive-success hover:border-interactive-success-hovered hover:focus-within:border-interactive-success":
                    success && isActive,
                  "border-critical focus-within:border-critical focus-within:ring-1 focus-within:ring-critical hover:border-interactive-critical-hovered hover:focus-within:border-critical":
                    error && isActive,
                  "border-none bg-transparent px-0 py-2 text-disabled hover:bg-transparent":
                    preview
                }
              )}
            >
              {prefix && (
                <span
                  className={cn(
                    "min-w-[20px] flex-shrink-0 whitespace-nowrap text-base font-normal leading-tight text-subdued",
                    {
                      "text-success": success && isActive,
                      "text-critical": error && isActive,
                      "text-disabled": disabled && !preview
                    }
                  )}
                >
                  {prefix}
                </span>
              )}
              {prefixIcon && (
                <div
                  className={cn(
                    "flex h-5 w-5 flex-shrink-0 items-center justify-center text-icon",
                    {
                      "text-icon-disabled": disabled && !preview,
                      "text-success": success && isActive,
                      "text-critical": error && isActive
                    }
                  )}
                >
                  {prefixIcon}
                </div>
              )}
              {inputField}
              {suffix && (
                <span
                  className={cn(
                    "min-w-[20px] flex-shrink-0 whitespace-nowrap text-base font-normal leading-tight text-subdued",
                    {
                      "text-success": success && isActive,
                      "text-critical": error && isActive,
                      "text-disabled": disabled && !preview
                    }
                  )}
                >
                  {suffix}
                </span>
              )}
              {charactersCountMarkup}
              {suffixIcon && (
                <div
                  className={cn(
                    "flex h-5 w-5 flex-shrink-0 items-center justify-center text-icon",
                    {
                      "text-icon-disabled": disabled && !preview,
                      "text-success": success && isActive,
                      "text-critical": error && isActive
                    }
                  )}
                >
                  {suffixIcon}
                </div>
              )}

              {stepperControlMarkup}
            </div>
            {hintMarkup}
            {messageMarkup}
          </div>
        )}
      </>
    );
  }
);

InputField.displayName = "InputField";
