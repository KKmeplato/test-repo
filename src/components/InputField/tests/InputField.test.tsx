import { fireEvent, render, screen } from "@testing-library/react";

import { InputField } from "../InputField";

const DummyPrefixIcon = () => <>SVGPrefixIcon</>;
const DummySuffixIcon = () => <>SVGSuffixIcon</>;

describe("<InputField />", () => {
  interface Props {
    label?: string;
    prefix?: string;
    suffix?: string;
    suffixIcon?: React.ReactElement;
    prefixIcon?: React.ReactElement;
    showCharacterCount?: boolean;
    valid?: boolean | undefined;
    hint?: string;
    showHintIcon?: boolean;
    errorMessage?: string;
    successMessage?: string;
    preview?: boolean;
    onChange?: () => void;
  }

  test("shows success messages", async () => {
    const { rerender } = render(
      <InputField
        id="test-text-input-field"
        type="text"
        label="Input text field"
        value="Test input text field"
        maxLength={255}
        hint="Lorem ipsum dolor sit amet, consetetur"
        data-testid="test-text-input"
        onChange={vi.fn()}
      />
    );

    const textInputElementWrapper =
      await screen.findByTestId("test-text-input");
    expect(textInputElementWrapper).toBeInTheDocument();
    expect(screen.queryByText("Error message")).toBeNull();
    expect(
      screen.queryByText("Lorem ipsum dolor sit amet, consetetur")
    ).not.toBeNull();
    expect(
      screen.getByDisplayValue("Test input text field")
    ).toBeInTheDocument();
    expect(screen.getByDisplayValue("Test input text field")).toHaveAttribute(
      "id",
      "test-text-input-field"
    );

    rerender(
      <InputField
        id="test-text-input-field"
        type="text"
        label="Input text field"
        value="Test input text field"
        maxLength={255}
        hint="Lorem ipsum dolor sit amet, consetetur"
        data-testid="test-text-input"
        onChange={vi.fn()}
        success="Success message"
        prefixIcon={<i className="fa-solid fa-circle-check" />}
        suffixIcon={<i className="fa-solid fa-circle-check" />}
      />
    );

    expect(screen.queryByText("Success message")).not.toBeNull();
    // Do not display hint if successMessage
    expect(
      screen.queryByText("Lorem ipsum dolor sit amet, consetetur")
    ).toBeNull();
  });

  test("shows error messages", async () => {
    const { rerender } = render(
      <InputField
        id="test-text-input-field"
        type="text"
        label="Input text field"
        value="Test input text field"
        maxLength={255}
        hint="Lorem ipsum dolor sit amet, consetetur"
        data-testid="test-text-input"
        onChange={vi.fn()}
      />
    );

    const textInputElementWrapper =
      await screen.findByTestId("test-text-input");
    expect(textInputElementWrapper).toBeInTheDocument();
    expect(screen.queryByText("Error message")).toBeNull();
    expect(
      screen.queryByText("Lorem ipsum dolor sit amet, consetetur")
    ).not.toBeNull();
    expect(
      screen.getByDisplayValue("Test input text field")
    ).toBeInTheDocument();
    expect(screen.getByDisplayValue("Test input text field")).toHaveAttribute(
      "id",
      "test-text-input-field"
    );

    rerender(
      <InputField
        id="test-text-input-field"
        type="text"
        label="Input text field"
        value="Test input text field"
        maxLength={255}
        hint="Lorem ipsum dolor sit amet, consetetur"
        data-testid="test-text-input"
        onChange={vi.fn()}
        error="Error message"
      />
    );

    expect(screen.queryByText("Error message")).not.toBeNull();
    // Do not display hint if successMessage
    expect(
      screen.queryByText("Lorem ipsum dolor sit amet, consetetur")
    ).toBeNull();
  });

  test("changes success to error message", async () => {
    const { rerender } = render(
      <InputField
        id="test-text-input-field"
        type="text"
        label="Input text field"
        value="Test input text field"
        maxLength={255}
        hint="Lorem ipsum dolor sit amet, consetetur"
        data-testid="test-text-input"
        success="Success message"
        onChange={vi.fn()}
      />
    );

    expect(screen.queryByText("Success message")).not.toBeNull();
    // Do not display hint if successMessage
    expect(
      screen.queryByText("Lorem ipsum dolor sit amet, consetetur")
    ).toBeNull();

    rerender(
      <InputField
        id="test-text-input-field"
        type="text"
        label="Input text field"
        value=""
        maxLength={255}
        hint="Lorem ipsum dolor sit amet, consetetur"
        data-testid="test-text-input"
        error="Input text field should not be empty!"
        onChange={vi.fn()}
      />
    );

    expect(
      screen.queryByText("Input text field should not be empty!")
    ).not.toBeNull();
    // Do not display hint if successMessage
    expect(
      screen.queryByText("Lorem ipsum dolor sit amet, consetetur")
    ).toBeNull();
  });

  test("shows hint text", async () => {
    const { rerender } = render(
      <InputField
        id="company-id"
        label={"Company name"}
        name="legalName"
        value={"Meplato"}
        maxLength={255}
        hint={"Meplato GmbH"}
        onChange={vi.fn()}
        data-testid="test-text-input"
      />
    );

    const labelElement = await screen.queryByText("Company name");
    expect(labelElement).toBeInTheDocument();

    const inputElement = await screen.findByTestId("test-text-input");
    expect(inputElement).toBeInTheDocument();

    const hintElement = await screen.queryByText("Meplato GmbH");
    expect(hintElement).toBeInTheDocument();
    expect(hintElement).toHaveTextContent("Meplato GmbH");

    rerender(
      <InputField
        id="company-id"
        label={"Company name"}
        name="legalName"
        value={"Meplato"}
        maxLength={255}
        onChange={vi.fn()}
        data-testid="test-text-input"
      />
    );

    /** Test no hint provided */
    expect(hintElement).not.toBeInTheDocument();
  });

  test("shows value or placeholder", async () => {
    const { rerender } = render(
      <InputField
        id="company-id"
        label={"Company name"}
        name="legalName"
        value={""}
        maxLength={255}
        placeholder={"Meplato GmbH"}
        onChange={vi.fn()}
        data-testid="test-text-input"
      />
    );

    const labelElement = await screen.queryByText("Company name");
    expect(labelElement).toBeInTheDocument();

    const inputElement = await screen.getByPlaceholderText("Meplato GmbH");
    expect(inputElement).toBeInTheDocument();
    expect(inputElement).toHaveAttribute("id", "company-id");
    expect(inputElement).toHaveDisplayValue("");

    rerender(
      <InputField
        id="company-id"
        label={"Company name"}
        name="legalName"
        value={"Meplato GmbH"}
        maxLength={255}
        placeholder={""}
        onChange={vi.fn()}
        data-testid="test-text-input"
      />
    );

    expect(inputElement).toHaveDisplayValue("Meplato GmbH");
  });

  test("changes value", async () => {
    const mockOnChange = vi.fn() as Props["onChange"];

    render(
      <InputField
        id="company-id"
        label={"Company name"}
        name="legalName"
        value={""}
        maxLength={255}
        onChange={mockOnChange}
        data-testid="test-text-input"
      />
    );

    const inputElement = await screen.getByTestId("test-text-input");
    expect(inputElement).toBeInTheDocument();
    expect(inputElement).toHaveTextContent("");
    fireEvent.change(inputElement, { target: { value: "Meplato test" } });
    expect(mockOnChange).toHaveBeenCalled();
  });

  test("renders a textarea when multiline prop is set", () => {
    const { container } = render(
      <InputField multiline={true} value="test" onChange={vi.fn()} />
    );
    const textareaElement = container.querySelector("textarea");
    const inputElement = container.querySelector("input");

    expect(textareaElement).toBeInTheDocument();
    expect(inputElement).not.toBeInTheDocument();
  });

  test("renders an input when there is no multiline prop", () => {
    const { container } = render(
      <InputField multiline={false} value="test" onChange={vi.fn()} />
    );
    const textareaElement = container.querySelector("textarea");
    const inputElement = container.querySelector("input");

    expect(inputElement).toBeInTheDocument();
    expect(textareaElement).not.toBeInTheDocument();
  });

  test("renders a `text` InputField", () => {
    const component = render(
      <InputField
        id="test-text-input-field"
        type="text"
        label="Input text field"
        value="Test input text field"
        maxLength={255}
        hint="Lorem ipsum dolor sit amet, consetetur"
        data-testid="test-text-input"
        onChange={vi.fn()}
        error="Error message"
        prefixIcon={<DummyPrefixIcon data-testid="test-prefix-icon" />}
        suffixIcon={<DummySuffixIcon data-testid="test-suffix-icon" />}
      />
    );
    expect(component).toMatchSnapshot();
  });

  describe("NumberInputField", () => {
    test("renders", () => {
      const component = render(
        <InputField
          type="number"
          id="test-number-input-field"
          label="Input text field"
          value={6}
          min={0}
          max={10}
          hint="Lorem ipsum dolor sit amet, consetetur"
          data-testid="test-text-input"
          onChange={vi.fn()}
        />
      );
      expect(component).toMatchSnapshot();
    });

    test("renders a number spinner", () => {
      const { getByTestId } = render(
        <InputField
          type="number"
          id="test-number-input-field"
          label="Input text field"
          value={6}
          min={0}
          max={10}
          hint="Lorem ipsum dolor sit amet, consetetur"
          data-testid="test-text-input"
          onChange={vi.fn()}
        />
      );

      const incrementButton = getByTestId("increment");
      const decrementButton = getByTestId("decrement");

      expect(incrementButton).toBeDefined();
      expect(incrementButton).toHaveAttribute("role", "button");

      expect(decrementButton).toBeDefined();
      expect(decrementButton).toHaveAttribute("role", "button");
    });
  });
});
