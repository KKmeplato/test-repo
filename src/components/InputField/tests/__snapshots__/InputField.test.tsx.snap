// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<InputField /> > NumberInputField > renders 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="inline-flex w-full flex-col items-start justify-start gap-1"
      >
        <label
          class="w-full text-sm font-medium leading-tight text"
          for="test-number-input-field"
        >
          Input text field
        </label>
        <div
          class="group inline-flex h-9 w-full items-center justify-start gap-1 rounded-level-2 border border-solid border-default bg-interactive-plain py-2 pl-3 pr-2 focus-within:border-focus focus-within:ring-1 focus-within:ring-focus hover:border-hovered hover:bg-interactive-plain-hovered hover:focus-within:border-focus hover:focus-within:bg-interactive-plain"
        >
          <input
            class="w-full rounded font-normal appearance-none text uikit-hide-number-spin-controls focus:outline-none text-base bg-transparent"
            data-testid="test-text-input"
            id="test-number-input-field"
            max="10"
            min="0"
            type="number"
            value="6"
          />
          <div
            class="relative left-[6px] flex flex-col gap-0.5"
          >
            <div
              aria-disabled="false"
              class="h-[14px] w-[18px] cursor-pointer rounded-level-3 bg-surface-neutral text-icon select-none group-hover:bg-surface-neutral-hovered group-hover:text-icon-hovered"
              data-testid="increment"
              role="button"
              tabindex="-1"
            >
              <svg
                class="rounded-level-3 active:bg-surface-neutral-pressed"
                fill="none"
                viewBox="0 0 18 14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5.90169 9H12.0983C12.8493 9 13.27 8.24649 12.806 7.7324L9.70769 4.29945C9.34735 3.90018 8.65265 3.90018 8.29231 4.29945L5.19399 7.7324C4.73001 8.24649 5.15069 9 5.90169 9Z"
                  fill="currentColor"
                />
              </svg>
            </div>
            <div
              aria-disabled="false"
              class="h-[14px] w-[18px] cursor-pointer rounded-level-3 bg-surface-neutral text-icon select-none group-hover:bg-surface-neutral-hovered group-hover:text-icon-hovered"
              data-testid="decrement"
              role="button"
              tabindex="-1"
            >
              <svg
                class="rounded-level-3 active:bg-surface-neutral-pressed"
                fill="none"
                viewBox="0 0 18 14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12.0983 5L5.90169 5C5.15069 5 4.73001 5.75351 5.19399 6.2676L8.29231 9.70055C8.65265 10.0998 9.34735 10.0998 9.70769 9.70055L12.806 6.2676C13.27 5.75351 12.8493 5 12.0983 5Z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="inline-flex gap-1"
        >
          <p
            class="shrink grow basis-0 text-sm font-normal leading-5 text-subdued"
          >
            Lorem ipsum dolor sit amet, consetetur
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="inline-flex w-full flex-col items-start justify-start gap-1"
    >
      <label
        class="w-full text-sm font-medium leading-tight text"
        for="test-number-input-field"
      >
        Input text field
      </label>
      <div
        class="group inline-flex h-9 w-full items-center justify-start gap-1 rounded-level-2 border border-solid border-default bg-interactive-plain py-2 pl-3 pr-2 focus-within:border-focus focus-within:ring-1 focus-within:ring-focus hover:border-hovered hover:bg-interactive-plain-hovered hover:focus-within:border-focus hover:focus-within:bg-interactive-plain"
      >
        <input
          class="w-full rounded font-normal appearance-none text uikit-hide-number-spin-controls focus:outline-none text-base bg-transparent"
          data-testid="test-text-input"
          id="test-number-input-field"
          max="10"
          min="0"
          type="number"
          value="6"
        />
        <div
          class="relative left-[6px] flex flex-col gap-0.5"
        >
          <div
            aria-disabled="false"
            class="h-[14px] w-[18px] cursor-pointer rounded-level-3 bg-surface-neutral text-icon select-none group-hover:bg-surface-neutral-hovered group-hover:text-icon-hovered"
            data-testid="increment"
            role="button"
            tabindex="-1"
          >
            <svg
              class="rounded-level-3 active:bg-surface-neutral-pressed"
              fill="none"
              viewBox="0 0 18 14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.90169 9H12.0983C12.8493 9 13.27 8.24649 12.806 7.7324L9.70769 4.29945C9.34735 3.90018 8.65265 3.90018 8.29231 4.29945L5.19399 7.7324C4.73001 8.24649 5.15069 9 5.90169 9Z"
                fill="currentColor"
              />
            </svg>
          </div>
          <div
            aria-disabled="false"
            class="h-[14px] w-[18px] cursor-pointer rounded-level-3 bg-surface-neutral text-icon select-none group-hover:bg-surface-neutral-hovered group-hover:text-icon-hovered"
            data-testid="decrement"
            role="button"
            tabindex="-1"
          >
            <svg
              class="rounded-level-3 active:bg-surface-neutral-pressed"
              fill="none"
              viewBox="0 0 18 14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12.0983 5L5.90169 5C5.15069 5 4.73001 5.75351 5.19399 6.2676L8.29231 9.70055C8.65265 10.0998 9.34735 10.0998 9.70769 9.70055L12.806 6.2676C13.27 5.75351 12.8493 5 12.0983 5Z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="inline-flex gap-1"
      >
        <p
          class="shrink grow basis-0 text-sm font-normal leading-5 text-subdued"
        >
          Lorem ipsum dolor sit amet, consetetur
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<InputField /> > renders a \`text\` InputField 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="inline-flex w-full flex-col items-start justify-start gap-1"
      >
        <label
          class="w-full text-sm font-medium leading-tight text"
          for="test-text-input-field"
        >
          Input text field
        </label>
        <div
          class="group inline-flex h-9 w-full items-center justify-start gap-1 rounded-level-2 border border-solid bg-interactive-plain py-2 pl-3 pr-2 hover:bg-interactive-plain-hovered hover:focus-within:bg-interactive-plain border-critical focus-within:border-critical focus-within:ring-1 focus-within:ring-critical hover:border-interactive-critical-hovered hover:focus-within:border-critical"
        >
          <div
            class="flex h-5 w-5 flex-shrink-0 items-center justify-center text-critical"
          >
            SVGPrefixIcon
          </div>
          <input
            class="w-full rounded font-normal appearance-none text uikit-hide-number-spin-controls focus:outline-none text-base bg-transparent text-critical"
            data-testid="test-text-input"
            id="test-text-input-field"
            maxlength="255"
            type="text"
            value="Test input text field"
          />
          <div
            class="flex h-5 w-5 flex-shrink-0 items-center justify-center text-critical"
          >
            SVGSuffixIcon
          </div>
        </div>
        <div
          class="inline-flex gap-2 text-critical"
        >
          <div
            class="inline-flex h-5 w-5 justify-center"
          >
            <svg
              fill="none"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 18C12.1217 18 14.1566 17.1571 15.6569 15.6569C17.1571 14.1566 18 12.1217 18 10C18 7.87827 17.1571 5.84344 15.6569 4.34315C14.1566 2.84285 12.1217 2 10 2C7.87827 2 5.84344 2.84285 4.34315 4.34315C2.84285 5.84344 2 7.87827 2 10C2 12.1217 2.84285 14.1566 4.34315 15.6569C5.84344 17.1571 7.87827 18 10 18ZM10 6C10.4156 6 10.75 6.33437 10.75 6.75V10.25C10.75 10.6656 10.4156 11 10 11C9.58437 11 9.25 10.6656 9.25 10.25V6.75C9.25 6.33437 9.58437 6 10 6ZM9 13C9 12.7348 9.10536 12.4804 9.29289 12.2929C9.48043 12.1054 9.73478 12 10 12C10.2652 12 10.5196 12.1054 10.7071 12.2929C10.8946 12.4804 11 12.7348 11 13C11 13.2652 10.8946 13.5196 10.7071 13.7071C10.5196 13.8946 10.2652 14 10 14C9.73478 14 9.48043 13.8946 9.29289 13.7071C9.10536 13.5196 9 13.2652 9 13Z"
                fill="currentColor"
              />
            </svg>
          </div>
          <div
            class="text-sm font-normal text-critical"
          >
            Error message
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="inline-flex w-full flex-col items-start justify-start gap-1"
    >
      <label
        class="w-full text-sm font-medium leading-tight text"
        for="test-text-input-field"
      >
        Input text field
      </label>
      <div
        class="group inline-flex h-9 w-full items-center justify-start gap-1 rounded-level-2 border border-solid bg-interactive-plain py-2 pl-3 pr-2 hover:bg-interactive-plain-hovered hover:focus-within:bg-interactive-plain border-critical focus-within:border-critical focus-within:ring-1 focus-within:ring-critical hover:border-interactive-critical-hovered hover:focus-within:border-critical"
      >
        <div
          class="flex h-5 w-5 flex-shrink-0 items-center justify-center text-critical"
        >
          SVGPrefixIcon
        </div>
        <input
          class="w-full rounded font-normal appearance-none text uikit-hide-number-spin-controls focus:outline-none text-base bg-transparent text-critical"
          data-testid="test-text-input"
          id="test-text-input-field"
          maxlength="255"
          type="text"
          value="Test input text field"
        />
        <div
          class="flex h-5 w-5 flex-shrink-0 items-center justify-center text-critical"
        >
          SVGSuffixIcon
        </div>
      </div>
      <div
        class="inline-flex gap-2 text-critical"
      >
        <div
          class="inline-flex h-5 w-5 justify-center"
        >
          <svg
            fill="none"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10 18C12.1217 18 14.1566 17.1571 15.6569 15.6569C17.1571 14.1566 18 12.1217 18 10C18 7.87827 17.1571 5.84344 15.6569 4.34315C14.1566 2.84285 12.1217 2 10 2C7.87827 2 5.84344 2.84285 4.34315 4.34315C2.84285 5.84344 2 7.87827 2 10C2 12.1217 2.84285 14.1566 4.34315 15.6569C5.84344 17.1571 7.87827 18 10 18ZM10 6C10.4156 6 10.75 6.33437 10.75 6.75V10.25C10.75 10.6656 10.4156 11 10 11C9.58437 11 9.25 10.6656 9.25 10.25V6.75C9.25 6.33437 9.58437 6 10 6ZM9 13C9 12.7348 9.10536 12.4804 9.29289 12.2929C9.48043 12.1054 9.73478 12 10 12C10.2652 12 10.5196 12.1054 10.7071 12.2929C10.8946 12.4804 11 12.7348 11 13C11 13.2652 10.8946 13.5196 10.7071 13.7071C10.5196 13.8946 10.2652 14 10 14C9.73478 14 9.48043 13.8946 9.29289 13.7071C9.10536 13.5196 9 13.2652 9 13Z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          class="text-sm font-normal text-critical"
        >
          Error message
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
